### ISSUE TITLE
Failing test[s] with Emacs 29.1

### ISSUE THREAD

**Issue by sten0** [zkry/yaml.el#51]
Hi @zkry,

In Debian we've encountered the following failing test, and it's a release critical bug for our copy of yaml.el.  Would you please take a look at the following when you have a moment?  This backtrace is from 0.5.3; however, the bug also exists in 0.5.2 (https://bugs.debian.org/1052947).

```elisp
Error: error ("Test ‘yaml-parsing-scalar’ redefined")
  mapbacktrace(#f(compiled-function (evald func args flags) #<bytecode 0x2bee69debae868e>))
  debug-early-backtrace()
  debug-early(error (error "Test ‘yaml-parsing-scalar’ redefined"))
  error("Test `%s' redefined" yaml-parsing-scalar)
  ert-set-test(yaml-parsing-scalar #s(ert-test :name yaml-parsing-scalar :documentation "Test conversion of !!str to JSON scalar" :body (closure (t) nil (let* ((fn-597 #'equal) (args-598 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list :null (yaml--resolve-scalar-tag "null"))) (error (progn (setq fn-597 #'signal) (list (car err) (cdr err))))))) (let ((value-599 'ert-form-evaluation-aborted-600)) (let (form-description-601) (if (unwind-protect (setq value-599 (apply fn-597 args-598)) (setq form-description-601 (nconc (list '(should (equal :null (yaml--resolve-scalar-tag "null")))) (list :form (cons fn-597 args-598)) (if (eql value-599 'ert-form-evaluation-aborted-600) nil (list :value value-599)) (if (eql value-599 'ert-form-evaluation-aborted-600) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-598)) nil))))) (ert--signal-should-execution form-description-601)) nil (ert-fail form-description-601))) value-599)) (let* ((fn-602 #'equal) (args-603 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list :false (yaml--resolve-scalar-tag "false"))) (error (progn (setq fn-602 #'signal) (list (car err) (cdr err))))))) (let ((value-604 'ert-form-evaluation-aborted-605)) (let (form-description-606) (if (unwind-protect (setq value-604 (apply fn-602 args-603)) (setq form-description-606 (nconc (list '(should (equal :false (yaml--resolve-scalar-tag "false")))) (list :form (cons fn-602 args-603)) (if (eql value-604 'ert-form-evaluation-aborted-605) nil (list :value value-604)) (if (eql value-604 'ert-form-evaluation-aborted-605) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-603)) nil))))) (ert--signal-should-execution form-description-606)) nil (ert-fail form-description-606))) value-604)) (let* ((fn-607 #'equal) (args-608 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list t (yaml--resolve-scalar-tag "true"))) (error (progn (setq fn-607 #'signal) (list (car err) (cdr err))))))) (let ((value-609 'ert-form-evaluation-aborted-610)) (let (form-description-611) (if (unwind-protect (setq value-609 (apply fn-607 args-608)) (setq form-description-611 (nconc (list '(should (equal t (yaml--resolve-scalar-tag "true")))) (list :form (cons fn-607 args-608)) (if (eql value-609 'ert-form-evaluation-aborted-610) nil (list :value value-609)) (if (eql value-609 'ert-form-evaluation-aborted-610) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-608)) nil))))) (ert--signal-should-execution form-description-611)) nil (ert-fail form-description-611))) value-609)) (let* ((fn-612 #'equal) (args-613 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list "xtruex" (yaml--resolve-scalar-tag "xtruex"))) (error (progn (setq fn-612 #'signal) (list (car err) (cdr err))))))) (let ((value-614 'ert-form-evaluation-aborted-615)) (let (form-description-616) (if (unwind-protect (setq value-614 (apply fn-612 args-613)) (setq form-description-616 (nconc (list '(should (equal "xtruex" (yaml--resolve-scalar-tag "xtruex")))) (list :form (cons fn-612 args-613)) (if (eql value-614 'ert-form-evaluation-aborted-615) nil (list :value value-614)) (if (eql value-614 'ert-form-evaluation-aborted-615) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-613)) nil))))) (ert--signal-should-execution form-description-616)) nil (ert-fail form-description-616))) value-614)) (let* ((fn-617 #'equal) (args-618 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list 0 (yaml--resolve-scalar-tag "0"))) (error (progn (setq fn-617 #'signal) (list (car err) (cdr err))))))) (let ((value-619 'ert-form-evaluation-aborted-620)) (let (form-description-621) (if (unwind-protect (setq value-619 (apply fn-617 args-618)) (setq form-description-621 (nconc (list '(should (equal 0 (yaml--resolve-scalar-tag "0")))) (list :form (cons fn-617 args-618)) (if (eql value-619 'ert-form-evaluation-aborted-620) nil (list :value value-619)) (if (eql value-619 'ert-form-evaluation-aborted-620) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-618)) nil))))) (ert--signal-should-execution form-description-621)) nil (ert-fail form-description-621))) value-619)) (let* ((fn-622 #'equal) (args-623 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list 10 (yaml--resolve-scalar-tag "10"))) (error (progn (setq fn-622 #'signal) (list (car err) (cdr err))))))) (let ((value-624 'ert-form-evaluation-aborted-625)) (let (form-description-626) (if (unwind-protect (setq value-624 (apply fn-622 args-623)) (setq form-description-626 (nconc (list '(should (equal 10 (yaml--resolve-scalar-tag "10")))) (list :form (cons fn-622 args-623)) (if (eql value-624 'ert-form-evaluation-aborted-625) nil (list :value value-624)) (if (eql value-624 'ert-form-evaluation-aborted-625) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-623)) nil))))) (ert--signal-should-execution form-description-626)) nil (ert-fail form-description-626))) value-624)) (let* ((fn-627 #'equal) (args-628 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list "x10" (yaml--resolve-scalar-tag "x10"))) (error (progn (setq fn-627 #'signal) (list (car err) (cdr err))))))) (let ((value-629 'ert-form-evaluation-aborted-630)) (let (form-description-631) (if (unwind-protect (setq value-629 (apply fn-627 args-628)) (setq form-description-631 (nconc (list '(should (equal "x10" (yaml--resolve-scalar-tag "x10")))) (list :form (cons fn-627 args-628)) (if (eql value-629 'ert-form-evaluation-aborted-630) nil (list :value value-629)) (if (eql value-629 'ert-form-evaluation-aborted-630) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-628)) nil))))) (ert--signal-should-execution form-description-631)) nil (ert-fail form-description-631))) value-629)) (let* ((fn-632 #'equal) (args-633 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list 10.52 (yaml--resolve-scalar-tag "10.52"))) (error (progn (setq fn-632 #'signal) (list (car err) (cdr err))))))) (let ((value-634 'ert-form-evaluation-aborted-635)) (let (form-description-636) (if (unwind-protect (setq value-634 (apply fn-632 args-633)) (setq form-description-636 (nconc (list '(should (equal 10.52 (yaml--resolve-scalar-tag "10.52")))) (list :form (cons fn-632 args-633)) (if (eql value-634 'ert-form-evaluation-aborted-635) nil (list :value value-634)) (if (eql value-634 'ert-form-evaluation-aborted-635) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-633)) nil))))) (ert--signal-should-execution form-description-636)) nil (ert-fail form-description-636))) value-634)) (let* ((fn-637 #'equal) (args-638 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list 52.0 (yaml--resolve-scalar-tag ".52e2"))) (error (progn (setq fn-637 #'signal) (list (car err) (cdr err))))))) (let ((value-639 'ert-form-evaluation-aborted-640)) (let (form-description-641) (if (unwind-protect (setq value-639 (apply fn-637 args-638)) (setq form-description-641 (nconc (list '(should (equal 52.0 (yaml--resolve-scalar-tag ".52e2")))) (list :form (cons fn-637 args-638)) (if (eql value-639 'ert-form-evaluation-aborted-640) nil (list :value value-639)) (if (eql value-639 'ert-form-evaluation-aborted-640) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (applyTest ‘yaml-parsing-scalar’ redefined
 -explainer- args-638)) nil))))) (ert--signal-should-execution form-description-641)) nil (ert-fail form-description-641))) value-639)) (let* ((fn-642 #'equal) (args-643 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list 1.0e+INF (yaml--resolve-scalar-tag ".Inf"))) (error (progn (setq fn-642 #'signal) (list (car err) (cdr err))))))) (let ((value-644 'ert-form-evaluation-aborted-645)) (let (form-description-646) (if (unwind-protect (setq value-644 (apply fn-642 args-643)) (setq form-description-646 (nconc (list '(should (equal 1.0e+INF (yaml--resolve-scalar-tag ".Inf")))) (list :form (cons fn-642 args-643)) (if (eql value-644 'ert-form-evaluation-aborted-645) nil (list :value value-644)) (if (eql value-644 'ert-form-evaluation-aborted-645) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-643)) nil))))) (ert--signal-should-execution form-description-646)) nil (ert-fail form-description-646))) value-644)) (let* ((fn-647 #'equal) (args-648 (condition-case err (let ((signal-hook-function #'ert--should-signal-hook)) (list "hello world" (yaml--resolve-scalar-tag "hello world"))) (error (progn (setq fn-647 #'signal) (list (car err) (cdr err))))))) (let ((value-649 'ert-form-evaluation-aborted-650)) (let (form-description-651) (if (unwind-protect (setq value-649 (apply fn-647 args-648)) (setq form-description-651 (nconc (list '(should (equal "hello world" (yaml--resolve-scalar-tag "hello world")))) (list :form (cons fn-647 args-648)) (if (eql value-649 'ert-form-evaluation-aborted-650) nil (list :value value-649)) (if (eql value-649 'ert-form-evaluation-aborted-650) nil (let* ((-explainer- (and t (ert--get-explainer 'equal)))) (if -explainer- (list :explanation (apply -explainer- args-648)) nil))))) (ert--signal-should-execution form-description-651)) nil (ert-fail form-description-651))) value-649))) :most-recent-result nil :expected-result-type :passed :tags nil :file-name "/<<PKGBUILDDIR>>/yaml-tests.el"))
  load-with-code-conversion("/<<PKGBUILDDIR>>/yaml-tests.el" "/<<PKGBUILDDIR>>/yaml-tests.el" nil t)
  command-line-1(("-l" "package" "--eval" "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa\")" "--eval" "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa-src\")" "-f" "package-initialize" "-L" "." "--eval" "(load-file \"yaml-tests.el\")" "-l" "yaml-tests.el" "--eval" "(ert-run-tests-batch-and-exit)"))
  command-line()
  normal-top-level()
dh_elpa_test: error: emacs -batch -Q -l package --eval "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa\")" --eval "(add-to-list 'package-directory-list \"/usr/share/emacs/site-lisp/elpa-src\")" -f package-initialize -L . --eval "(load-file \"yaml-tests.el\")" -l yaml-tests.el --eval \(ert-run-tests-batch-and-exit\) returned exit code 255
```

**Comment by zkry** [comment]
Thanks for the report! It looks like the parsing state variables weren't being initialized for the test. I'll work on fixing this.

**Comment by sten0** [comment]
Zachary Romero ***@***.***> writes:

> Thanks for the report! It looks like the parsing state variables weren't being initialized for the test. I'll work on fixing this. 
>


Thanks, much appreciated!

**Comment by sten0** [comment]
Zachary Romero ***@***.***> writes:

> Closed #51 as completed via #52.

Wow, that was fast!  Much obliged! :)  Likewise, the fix has now been
uploaded to Debian.
